src/geo/io.clj

```diff
@@ -128,14 +128,14 @@
 
 (defn to-geojson [shapelike] (.toString (.write (GeoJSONWriter.) (to-jts shapelike jts/default-srid))))
 
-(defn- ^Feature gj-feature
-  [{shapelike :geometry properties :properties}]
-  (let [gj-geom (.write (GeoJSONWriter.) (to-jts shapelike jts/default-srid))]
-    (Feature. gj-geom (stringify-keys properties))))
+(defn- ^org.wololo.geojson.Feature gj-feature
```

Comment:
1 more question -- what happens if you call this with a map now? will it get coerced into a feature, or error in some way?

Reply:
It tries to coerce it into a feature. If either of those two fields are not included in the map, those will basically be created with `nil` values, though a nil geometry will cause the function to fail, because `to-jts` doesn't know how to operate on `nil`. Overall, it seems like good practice (https://stuartsierra.com/2015/05/17/clojure-record-constructors) to create constructor functions for records, but that seems like it would involve making a lot more decisions about exactly what kinds of things could or couldn't be passed into geometry or properties.

```clojure
(gj-feature {:geometry (jts/point 10 10)})
=>
#object[org.wololo.geojson.Feature
        0x40717620
        "{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[10.0,10.0]},\"properties\":null}"]
```

Reply:
Got it. For the current behavior, it seems reasonable to expect `gj-feature` without a geometry to fail, right? And I guess having properties get nulled out is probably OK to. So I think we're happy with how it is working now?

Reply:
I agree, I think this is the right move for now. Just the existence of a record that will always have a placeholder for geometry and properties is, I think, enough of a signpost to start. Many of the other functions here fail with bad input, so I don't think this is out of the ordinary. After some more thought we could consider eventually creating a constructor that makes sure that the geometry field is a valid shapelike and that the properties are properly keywordized and all that, but I don't have enough variable real-world workflows using all this just yet to be sure that the constructor would be appropriately strict or forgiving.